erigon-lib/kv/kvcache/cache_test.go

```diff
@@ -173,13 +172,19 @@ func TestEviction(t *testing.T) {
 }
 
 func TestAPI(t *testing.T) {
-	if runtime.GOOS != "linux" {
-		t.Skip("fix me on win please")
-	}
 	require := require.New(t)
+
+	// Create a context with timeout for the entire test
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
 	c := New(DefaultCoherentConfig)
 	k1, k2 := [20]byte{1}, [20]byte{2}
 	db, _ := temporaltest.NewTestDB(t, datadir.New(t.TempDir()))
+
+	// Ensure proper cleanup
+	defer db.Close()
```

Comment:
look inside `temporaltest.NewTestDB` there is 


```
if tb != nil {
		tb.Cleanup(agg.Close)
	}
```
so, there is no need to cleanup outside. 

Ah! Seems there is no `tb.Cleanup(db.Close)` oops - it's a bug. fixing by: https://github.com/erigontech/erigon/pull/14513